src/shims/mod.rs

```diff
@@ -27,9 +27,25 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         // There are some more lang items we want to hook that CTFE does not hook (yet).
         if this.tcx.lang_items().align_offset_fn() == Some(instance.def.def_id()) {
-            // FIXME: return a real value in case the target allocation has an
-            // alignment bigger than the one requested.
-            let n = u128::max_value();
+
+            let n = {
+                let ptr = this.force_ptr(this.read_scalar(args[0])?.not_undef()?)?;
+                let align = this.force_bits(
+                    this.read_scalar(args[1])?.not_undef()?,
+                    this.pointer_size()
+                )? as usize;
+
+                let stride = this.memory().get(ptr.alloc_id)?.align.bytes() as usize;
+                // if the allocation alignment is at least the required alignment, we use the
+                // libcore implementation
+                if stride >= align {
+                    ((stride + ptr.offset.bytes() as usize) as *const ())
```

Comment:
This is interesting. Please add a comment (in a new PR, I guess) explaining why it is correct. After all, you are not using the "true" base address of the allocation... so why `stride + ptr.offset` and not just `ptr.offset`?

Reply:
Seems overkill to use the true base address if this yields the same result except for the overflowing UB

Reply:
What I am asking for is an argument for *why* this yields the same result (and why the overflowing difference is okay).

Also, I think the `stride +` is entirely unnecessary.

Reply:
Though TBH I think it is simpler to just do a ptr-to-int cast on `ptr` and use that, instead of trying to be clever here. What is gained by the current approach?